synapse/storage/database.py

```diff
@@ -121,6 +153,7 @@ def make_pool(
     # someone has explicitly set `cp_reconnect`.
     db_args = dict(db_config.config.get("args", {}))
     db_args.setdefault("cp_reconnect", True)
```

Comment:
Can you please update the description with more information on what this PR is attempting to do. From the current code it seems that there's some assumption of re-connection support already.